packages/wrangler/src/deployment-bundle/bindings.ts

```diff
@@ -268,67 +321,90 @@ async function runProvisioningFlow(
 
 		for (const item of pending) {
 			logger.log("Provisioning", item.binding, `(${friendlyBindingName})...`);
-			let name: string = "";
+			let name = item.name;
 			let selected: string;
 
-			if (options.length === 0 || autoCreate) {
-				selected = NEW_OPTION_VALUE;
-			} else {
-				selected = await select(
-					`Would you like to connect an existing ${friendlyBindingName} or create a new one?`,
-					{
-						choices: options.concat([{ title: "Create new", value: "new" }]),
-						defaultOption: options.length,
+			if (name) {
+				logger.log("Resource name found in config:", name);
+				const foundResourceId = preExisting.find(
+					(r) => r.title === name
+				)?.value;
```

Comment:
What case is this catering for? Surely if the item has a name (D1 or R2) which maps to an existing resource we should have already bound to it further up? Why do we need to ask the user?